src/neo/SmartContract/ApplicationEngine.Contract.cs

```diff
@@ -71,14 +72,14 @@ partial class ApplicationEngine
         /// <param name="args">The arguments to be used.</param>
         protected internal void CallContract(UInt160 contractHash, string method, CallFlags callFlags, Array args)
         {
-            if (method.StartsWith('_')) throw new ArgumentException($"Invalid Method Name: {method}");
+            if (method.StartsWith('_')) throw new CatchableException($"Invalid method name: {method}");
```

Comment:
Some contract is trying to call `_method` of another contract, which means that either it's buggy or someone is trying to play some trick with it. Why should it be allowed to catch this exception? Failing completely is a better behavior.

And it's the same for inappropriate flags or missing contract.